sphinx_doc/tutorials/docs/writing_new_nav2planner_plugin.rst

```diff
@@ -0,0 +1,215 @@
+.. _writing_new_nav2planner_plugin:
+
+Writing a new planner plugin
+****************************
+
+- `Overview`_
+- `Requirements`_
+- `Tutorial Steps`_
+
+Overview
+========
+
+This tutorial shows how to create you own planner plugin.
+
+Requirements
+============
+
+- ROS2 (binary or build-from-source)
+- Navigation2 (Including dependencies)
+- Gazebo
+- Turtlebot3
+
+Tutorial Steps
+==============
+
+1- Creating a new Planner Plugin
+--------------------------------
+
+We will create a simple straight line planner.
+The annotated code in this tutorial can be found in `navigation_tutorials <https://github.com/ros-planning/navigation2_tutorials>`_ repository as the ``nav2_rrtconnect_plugin``
```

Comment:
You should rename `nav2_rrtconnect_plugin` -> `nav2_straightline_planner` or whatever you'd like to name it